packages/next-codemod/transforms/lib/async-request-api/utils.ts

```diff
@@ -492,3 +492,111 @@ export function getVariableDeclaratorId(
   }
   return undefined
 }
+
+export function findFunctionBody(path: ASTPath<FunctionScope>): null | any[] {
+  let functionBody = path.node.body
+  if (functionBody && functionBody.type === 'BlockStatement') {
+    return functionBody.body
+  }
+  return null
+}
+
+export const isReactHookName = (name: string) =>
+  // function name is `use`
+  name === 'use' ||
+  // function name is `useX*`
+  (name.startsWith('use') && name[3] === name[3].toUpperCase())
+
+// Determine a path of function contains any React hooks call expressions.
+// e.g. if there's any of those call expressions in the function body:
+// use() => true
+// React.use() => false
+// useXxxx() => true
+// Foo.use() => true
+// Foo.useXxxx() => true
+export function containsReactHooksCallExpressions(
+  path: ASTPath<FunctionScope>,
+  j: API['jscodeshift']
+) {
+  const hasReactHooks =
+    j(path)
+      .find(j.CallExpression)
+      .filter((callPath) => {
+        // Is it matching use(<callPath>)
+        // Is it matching useX*(<callPath>)
+        const isUseHookOrReactHookCall =
+          j.Identifier.check(callPath.value.callee) &&
+          isReactHookName(callPath.value.callee.name)
+
+        // is it matching member access React.use(<callPath>)
+        // is it matching member access <Any>.useX*(<callPath>)
```

Comment:
Must check for Uppercase.useX like `eslint-plugin-react-hooks` (e.g. `jest.useFakeTimers` should not be considered a Hook). You don't need to check for `React` specifically and can just reuse the Uppercase check i.e. the object name must start with an uppercase letter. If it's starts with a lowercase letter, it won't be an object whose properties are Hooks.

Reference impl: https://github.com/facebook/react/blob/09111202d617477b63507b41e8b6c3101b4afd87/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js#L22-L41

Reply:
It's not checking `React` specifically, those comments are examples, like will match the react one, also the common one.
The code below only check the member expression property if it's matching `.use*`

Reply:
I think I mixed it up with an earlier version of this branch.

Reply:
We have a `Foo.use` case in `packages/next-codemod/transforms/__testfixtures__/next-async-request-api-dynamic-props/access-props-36.input.tsx` now already, I can get a Foo.useX* as well